syntax = "proto3";

package rpc_package;

message Empty {}

service TreeService {
    rpc add_note_root (rootAddRequest) returns (rootAddReply) {}
    //streaming functions
    rpc Stream (Empty) returns (stream streamRequest) {}
    //rpc add_stream (streamRequest) returns (Empty) {}
}

message insertRequest {
    string client = 1;
}

message insertReply {
    bool status = 1;
    float minsup = 2;
    string message = 3;
}

message rootAddRequest {
    string client = 1;
    string trx = 2;
    string message = 3;
}

message rootAddReply {
    bool status = 1;
    string client = 2;
    string message = 3;
    string trx = 4;
}

message streamRequest {
    string client = 1;
    string trx = 2;
    string message = 3;
}